pennylane/devices/autograd_ops.py

```diff
@@ -180,3 +180,65 @@ def MultiRZ(theta, n):
         array[complex]: diagonal part of the multi-qubit rotation matrix
     """
     return np.exp(-1j * theta / 2 * pauli_eigs(n))
+
+
+def DoubleExcitation(phi):
+    r"""
+    Double excitation rotation.
+    Args:
+        phi (float): rotation angle
+    Returns:
+        array[complex]: Double excitation rotation matrix
+    """
+    c = np.cos(phi / 2)
+    s = np.sin(phi / 2)
+
+    U = np.zeros((16, 16))
+    U[3, 3] = c  # 3 (dec) = 0011 (bin)
+    U[3, 12] = -s  # 12 (dec) = 1100 (bin)
+    U[12, 3] = s
+    U[12, 12] = c

```

Comment:
Might be important to check that this remains differentiable --- I think autograd (and maybe JAX) do not like setting array elements via indexing

Reply:
Thanks for the suggestion! How would that check look like exactly? :thinking: 

Is this not checked in the device tests?

Reply:
Ah, so what you will need to do is just add a new integration test, which

* Creates a QNode parametrized on the diff method (both parameter-shift and backprop)
* Uses the new gate
* Differentiates the QNode and compares it against the expected result 🙂 

If everything is fine, the tests will pass for both diff methods.